docs/ingestion/compaction.md

```diff
@@ -22,36 +22,45 @@ description: "Defines compaction and automatic compaction (auto-compaction or au
   ~ specific language governing permissions and limitations
   ~ under the License.
   -->
-Query performance in Apache Druid depends on optimally sized segments. Compaction is one strategy you can use to optimize segment size for your Druid database. Compaction tasks read an existing set of segments for a given time interval and combine the data into a new "compacted" set of segments. The compacted segments are generally larger, but there are fewer of them. Here compaction increases performance because fewer segments require less the per-segment processing and the memory overhead for ingestion and for querying paths.
+Query performance in Apache Druid depends on optimally sized segments. Compaction is one strategy you can use to optimize segment size for your Druid database. Compaction tasks read an existing set of segments for a given time interval and combine the data into a new "compacted" set of segments. In some cases the compacted segments are larger, but there are fewer of them. In other cases the compacted segments may be smaller. Compaction tends to increase performance because optimized segments require less per-segment processing and less memory overhead for ingestion and for querying paths.
 
-As a general strategy, compaction is effective when you have data arriving out of chronological order resulting in lots of small segments. This often happens, for example, if you are appending data using `appendToExisting` for [native batch](./native_batch.md) ingestion. Conversely, if you are rewriting your data with each ingestion task, you don't need to use compaction.
+## Compaction strategies
+There are several cases to consider compaction for segment optimization:
+- With streaming ingestion, data can arrive out of chronological order creating lots of small segments.
+- If you append data using `appendToExisting` for [native batch](native-batch.md) ingestion creating suboptimal segments.
+- When you use `index_parallel` for parallel batch indexing and the parallel ingestion tasks create many small segments.
+- When a misconfigured ingestion task creates oversized segments.
 
-In some cases you can use compaction to reduce segment size. For example, if a misconfigured ingestion task creates oversized segments, you can create a compaction task to split the segment files into smaller, more optimally sized ones.
+By default, compaction does not modify the underlying data of the segments. However, there are cases when you may want to modify data during compaction to improve query performance:
+- If, after ingestion, you realize realize that data for the time interval is sparse, you can use compaction to increase the segment granularity.
+- Over time you don't need fine-grained granularity for older data so you use compaction to change older segments to a coarser query granularity. For example from `minute` to `hour`, or `hour` to `day`. You cannot go from coarser granularity to finer granularity.
```

Comment:
Should mentioned that the reason for this is for more rollup which result in less storage space